content/bd/0001-intro-dbms.md

```diff
@@ -0,0 +1,253 @@
+Course (course_id: string, course_name: string,
+        description: integer)
+Shift (shift_id: string, course_id: string, capacity: integer)
+Enrollment (ist_id: string, shift_id: string)
+Room (room_name: string, course_id: string, capacity: integer)
+```
+
+Uma possível _view_ para este esquema lógico seria:
+
+```
+CourseShiftRooms (room_name: string, course_name: string,
+                  enrolled_count: integer)
+```
+
+Podemos reparar que esta relação não está representada explicitamente no
+esquema lógico, no entanto todos os dados podem ser derivados das relações
+definidas no mesmo.
+
+:::
+
+A [**grande vantagem**](color:green) desde modelo é a independência dos dados. Como estamos
+a dividir a estrutura em três níveis, podemos alterar cada um deles sem ter de propagar
+estas alterações "para cima":
+
+- Podemos alterar o esquema lógico sem alterar as nossas aplicações, visto que as vistas
+  se podem manter.
+- Podemos alterar a organização física dos dados, isto é, como são armazenados, sem termos
+  de modificar o esquema lógico.
+
+## Modelos de Dados
+
+Um SGBD tem de usar um **modelo de dados**, um conjunto de conceitos e métodos, para descrever
+**dados**, **relacionamentos**, **semântica de dados** e **restrições**.
+
+Existem vários tipos de modelos de dados hoje em dia:
+
+- **Relacional** (o mais usado)
+- **Entidade-Relação** (para conceção de bases de dados)
+- **Orientada a Objetos**
+- **Semi-Estruturados** (XML, JSON, etc.)
+
+### Modelo Relacional
+
+Atualmente, o **modelo relacional** é o mais usado.
+Como o próprio nome indica, é composto por relações.
+Uma relação (chamada no dia-a-dia de tabela/_table_), é constituída pelo seu **nome**,
+o **nome de cada atributo** (_field_) e o **tipo de cada atributo**.
+
+:::info[Exemplo]
+
+```
+Student (ist_id: string, student_name: string, birthday: date)
+```
+
+Podemos depois adicionar linhas (_rows_) nesta relação. Neste caso, cada linha
+representa um único aluno.
+
+| `ist_id`   | `student_name` | `birthday` |
+| ---------- | -------------- | ---------- |
+| ist1123456 | Diogo          | 2002-03-06 |
+| ist1123234 | Rafa           | 2002-08-05 |
+| ist1124453 | Tomás          | 2002-12-18 |
+| ist1123534 | Diogo          | 2001-09-16 |
+| ist1123532 | Diogo          | 2002-04-27 |
+
+:::
+
+Neste exemplo podemos querer também enforçar restrições de integridade.
+Neste caso, a restrição mais óbvia é garantir que não existem dois alunos diferentes
+com o mesmo `ist_id`. No entanto, podem existir alunos com o mesmo nome.
+
+## Arquitetura de SGBDs
+
+Para falarmos com os SGBDs temos de usar uma linguagem. No caso do modelo relacional,
+essa linguagem é o [**SQL**](color:orange) (Structured Query Language).
+
+Na imagem abaixo podemos ver o esquema que representa a estrutura todo o fluxo de execução
+entre a aplicação e a base de dados, passando pelo SGBD.
+
+![Esquema que representa a arquitetura de um SGBD](./assets/0001-dbms-architecture.svg)
```

Comment:
```suggestion
![Esquema que representa a arquitetura de um SGBD](./assets/0001-dbms-architecture.svg#dark=3)
```
Black arrows on dark page background don't work.

Reply:
I was afraid it would mess up with the colors too much, but it actually looks decent